packages/kbn-config-schema/src/oas_meta_fields.ts

```diff
@@ -0,0 +1,17 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+/**
+ * These fields are used in Joi meta to capture additional information used
+ * by OpenAPI spec generator.
+ */
+export const META_FIELD_X_OAS_OPTIONAL = 'x-oas-optional' as const;
```

Comment:
the `x-` prefix is expected by our partner lib https://www.npmjs.com/package/joi-to-json (to be introduced in a follow up PR)